Cannot access certain globalThis properties in .ts files

As discussed in #650 and #2368, in Deno `window === globalThis`. This can be verified by running the statement in the REPL.

However, if you create a file...

```ts
// test.ts
console.log(window === globalThis);
```
and run `deno test.ts`, the tsc will throw an error, complaining that:
> error TS2367: This condition will always return 'false' since the types 'typeof globalThis' and 'Window' have no overlap.

This also prevents the ability to use things like `globalThis.fetch`, `globalThis.addEventListener`, and `globalThis.onload` in .ts files, even though this should be perfectly safe in Deno given that all these things work in ordinary .js files and the REPL.

Debug Info
Tested on Deno v0.12.0 running on Windows 10

--- a/cli/tests/types.out
+++ b/cli/tests/types.out
@@ -10,5 +10,5 @@ declare interface Window {
   Deno: typeof Deno;
 }
 
-declare const window: Window;
+declare const window: Window & typeof globalThis;
 [WILDCARD]

--- a/js/globals_test.ts
+++ b/js/globals_test.ts
@@ -14,7 +14,6 @@ test(function windowWindowExists(): void {
 });
 
 test(function globalThisEqualsWindow(): void {
-  // @ts-ignore (TypeScript thinks globalThis and window don't match)
   assert(globalThis === window);
 });
 

--- a/js/lib.deno_runtime.d.ts
+++ b/js/lib.deno_runtime.d.ts
@@ -1217,7 +1217,7 @@ declare namespace Deno {
 // @url js/globals.ts
 
 declare interface Window {
-  window: Window;
+  window: Window & typeof globalThis;
   atob: typeof textEncoding.atob;
   btoa: typeof textEncoding.btoa;
   fetch: typeof fetchTypes.fetch;
@@ -1263,7 +1263,7 @@ declare interface Window {
   Deno: typeof Deno;
 }
 
-declare const window: Window;
+declare const window: Window & typeof globalThis;
 declare const atob: typeof textEncoding.atob;
 declare const btoa: typeof textEncoding.btoa;
 declare const fetch: typeof fetchTypes.fetch;

